### PR TITLE
Deflector mass method

### PR DESCRIPTION
**Pull request by furcelay**
This pull request adds a new method `mass_model_lenstronomy` to the `DeflectorBase`, `EPLSersic`, `NFWHernquist`, `NFWCluster`, and `Deflector` classes. This method takes a _Lenstronomy_ `LensCosmo` instance and returns the mass profile parameters for the deflector in _Lenstronomy_ conventions.

Although this method depends on a source redshift through the `LensCosmo` instance, it is not tied to a specific source. The results are not stored within the deflector, ensuring that it adheres to the class definition: "Class of a single deflector with quantities only related to the deflector (independent of the source)."

The primary advantage of this addition is the simplification and generalization of the `deflector_mass_model_lenstronomy` method in the `Lens` class. This change removes the need for specific handling of each deflector type, especially the `NFWCluster`, which would otherwise require constructing models for the halo and each subhalo within the function (issue #156). By incorporating this new method, these operations are now managed within the `NFWCluster` and its subhalos, streamlining the process of implementing #134 and other new deflectors.

### CODE REVIEW

File: slsim/Deflectors/DeflectorTypes/deflector_base.py [hunk 1]
```diff
@@ -89,6 +89,16 @@ def mass_ellipticity(self):
         )
         return e1_mass, e2_mass
 
+    @abstractmethod
+    def mass_model_lenstronomy(self, lens_cosmo):
+        """Returns lens model instance and parameters in lenstronomy conventions.
+
+        :param lens_cosmo: lens cosmology model
+        :type lens_cosmo: LensCosmo class
```
Comment (sibirrer) [inline slsim/Deflectors/DeflectorTypes/deflector_base.py:97]: can you write this as ~lenstronomy.Cosmo.LensCosmo instance?

**Review by sibirrer — APPROVED** [review]
Thank you very much @furcelay! I like this change. It though requires the consistent treatment of the source redshift but I think this is a better way of doing it in the respective classes.
For multiple source redshift, the lenstronomy LensModel() class has a z_source_convention argument and this is the one for which the single-plane equivalent source redshifts are defined in the mass models (so that's what you then need to do, i.e. LensCosmo(z_source=z_source_convention). And then you can have multiple sources with different redshifts appended to it

File: slsim/Deflectors/DeflectorTypes/deflector_base.py (hunk 1 above)
Comment (furcelay) [reply slsim/Deflectors/DeflectorTypes/deflector_base.py]: Corrected in the last commit

**Comment by furcelay** [comment]
Thank you @sibirrer! I'll make sure to handle consistent treatment of the source redshift using `z_source_convention` in `LensCosmo` for multiple source cases.

**Review by nkhadka21 — APPROVED** [review]
@furcelay , thank you very much for This PR! I like these changes. I approve this PR. Looking forward to seeing your progress in  multiple source with different redshift case.

**Pull request merged**